src/dataflow/src/logging/materialized.rs

```diff
@@ -176,6 +197,32 @@ pub fn construct<A: Allocate>(
                                     ),
                                 }
                             }
+                            MaterializedEvent::Frontier(name, logical, delta) => {
+                                frontier_session.give((
+                                    row_packer.pack(&[
+                                        Datum::String(&name.to_string()),
+                                        Datum::Int64(worker as i64),
+                                        Datum::Int64(logical as i64),
+                                    ]),
+                                    time_ms,
+                                    delta as isize,
+                                ));
+                            }
+                            MaterializedEvent::KafkaConsumerInfo {
+                                source_name,
+                                source_id,
+                                consumer_name,
+                                rx,
+                                rx_bytes,
+                                tx,
+                                tx_bytes,
+                            } => {
+                                kafka_consumer_info_session.give((
+                                    (source_name, source_id, consumer_name),
+                                    time_ms,
+                                    DiffVector::new(vec![rx, rx_bytes, tx, tx_bytes]),
```

Comment:
Fwiw, this would be more efficiently implemented as a bunch of `DiffPair` things. It's a bunch less clean, but compiles down to a 4-tuple, rather than an allocation.

Reply:
When you say a bunch of `DiffPair` things, do you mean something like `DiffPair::new(DiffPair::new(rx, rx_bytes), DiffPair::new(tx, tx_bytes))`?

Reply:
Yeah, unfortunately.

Reply:
Sounds good to me! Thanks for the tip!